### ISSUE TITLE
Please add loopback.cfg

### ISSUE THREAD

**Issue by probonopd** [pop-os/iso#243]
**Issue/Bug Description:**

Please add a `loopback.cfg` like Ubuntu ISOs do. This will allow one to boot the ISO from hard disk in an easy way.

**Other Notes:**

https://www.supergrubdisk.org/wiki/Loopback.cfg

![](https://user-images.githubusercontent.com/2480569/32960053-e1b345d4-cbc3-11e7-9f34-c1cc11b5b1c7.png)

**Comment by probonopd** [comment]
Ran into this again today. Please add `loopback.cfg`.

**Comment by queeup** [comment]
No loopback on 20.04 either. :disappointed:

**Comment by space88man** [comment]
Pop_OS! ISO Casper doesn't containing parsing for `iso-scan/filename=`.

```
# on Pop_OS! initrd.gz
$ grep -r 'iso-scan' init scripts/
$

# on Ubuntu Desktop initrd.gz
$ grep -r 'iso-scan' init scripts/
scripts/casper-premount/20iso_scan:        iso-scan/filename=*)
scripts/casper-premount/20iso_scan:            iso_path=${x#iso-scan/filename=}
```

**Comment by probonopd** [comment]
Thanks for checking @space88man. Then it should be brought over from Ubuntu. It's in the package https://packages.ubuntu.com/eoan/lupin-casper.

**Comment by space88man** [comment]
Tested - I extracted the initrd.gz and add `/scripts/casper-premount/*` and  `/scripts/lupin-helpers` from Ubuntu desktop - then rebuilt the initrd.gz 

Instead of using GRUB's loopback, I pulled vmlinuz and the new initrd.gz to the HDD and booted from there.
Now iso-scan/filename= worked and Pop_OS! is able to run ISO-on-HDD.

TL;DR you need a new initrd.gz to run in ISO-on-HDD mode.

Off-topic: How did Pop_OS! 20.04 run on my Renoir APU when KDE Neon (latest 5.4.0) wouldn't and required `nomodeset`?

**Comment by jrudolph** [comment]
> TL;DR you need a slightly modified initrd.gz to run in ISO-on-HDD mode by adding casper's iso-scan scripts.

That works indeed. After adding `20isoscan` to `scripts/casper-premount`, I also had to add an entry to the `scripts/casper-premount/ORDER` file.

**Comment by probonopd** [comment]
So, can a `loopback.cfg` be added to the official ISOs please?

**Comment by jackpot51** [comment]
This seems like something any one of you could help fix with a PR. Add `lupin-casper` to the LIVE_PKGS in each `config/pop-os` file and create loopback.cfg in `mk/iso.mk`

**Comment by VPraharsha03** [comment]
> TL;DR you need a slightly modified initrd.gz to run in ISO-on-HDD mode by adding casper's iso-scan scripts.
> 
> Tested - I extracted `initrd.gz` and add `/scripts/casper-premount/*` and `/scripts/lupin-helpers` from the Ubuntu desktop ISO then rebuilt `initrd.gz`.
> 
> Instead of using GRUB's loopback, I pulled `vmlinuz` and the new `initrd.gz` to the HDD and referenced them in the GRUB menu entry in place of `(loop)/casper/vmlinuz` etc. Now
> 
> ```
> ... boot=casper iso-scan/filename=/pop-os_20.04_amd64_intel_13.iso uuid=2020-08-31-22-41-10-00 ...
> ```
> 
> works and Pop_OS! is able to run in ISO-on-HDD mode. I add the `uuid=` to be doubly sure, since my HDD has partitions with extracted ISOs for Ubuntu, KDE Neon, etc — didn't want casper to choose the wrong live image.
> 
> Off-topic: How did Pop_OS! 20.04 run on my Renoir APU when KDE Neon (latest 5.4.0) wouldn't and required `nomodeset`?

@space88man can you share the commands you used to extract & re-compress the gz because the gz file isn't a gzip file it seems (i'm doing it for pop os 19.04 

```gzip: INITRD.GZ: not in gzip format

initrd.gz: ASCII cpio archive (SVR4 with no CRC)```

surprisingly i can use WinRAR and then 7-zip to extract it's inner content

**Comment by VPraharsha03** [comment]
> Following commands assume the content of **initrd.gz** is (will be) in the current directory (taken from Slackware + [kernel.org](https://www.kernel.org/doc/html/latest/admin-guide/initrd.html)):
> 
> * **extract:** `gzip -cd /tmp/initrd.gz | cpio -imd`
> * **compress:** `find . | cpio -o -H newc | gzip -9c > /tmp/initrd-new.gz`

@Oifan Extraction follows a different sequence of cpio and gzip commands, but this can be avoided and directly `unmkinitramfs` can be used to extract everything in a go easily....

problem is some of the files do not get extracted (i've tried on different filesystems - NTFS & ext3 but they don't extract) - these files are from `lib\firmware\radeon\`

not sure if others have faced this issue or not

**Comment by space88man** [comment]
For reference Pop_OS! 21.04; build initrd with iso-scan capability:

```
# extract initrd, need to skip AMD(62 blocks)/Intel microcode(9004 blocks)
# YMMV as the block sizes will change
dd if=SRC/initrd.gz skip=9066 | cpio -imd

# now fixup scripts/casper-premount/{20iso_scan,ORDER} scripts/lupin-helpers
# from ubuntu 21.10 (has an extra 30xxxx script so adjust accordingly)

# reconstruct a initrd with microcode prepended
dd if=SRC/initrd.gz count=9066 > DEST/initrd-new.gz
find . | cpio -o -H newc | lz4 -l -c >> DEST/initrd-new.gz
```
